File: plugins/bridge/package/android/src/main/kotlin/gyo/plugins/bridge/BridgeRegistry.kt
package gyo.plugins.bridge

import android.util.Log

/**
 * Bridge Registry - Singleton for managing custom bridges
 */
object BridgeRegistry {
    private val handlers = mutableMapOf<String, BridgeHandler>()

    fun initialize() {
        handlers.clear()
    }

    fun register(bridgeName: String, handler: BridgeHandler) {
        handlers[bridgeName] = handler
        Log.d("BridgeRegistry", "Registered bridge: $bridgeName")
    }

    fun unregister(bridgeName: String) {
        handlers.remove(bridgeName)
        Log.d("BridgeRegistry", "Unregistered bridge: $bridgeName")
    }

    fun get(bridgeName: String): BridgeHandler? {
        return handlers[bridgeName]
    }
}

File: cli/templates/ios/Sources/GyoBridge/BridgeRegistry.swift
import Foundation

// MARK: - Bridge Registry

@MainActor
public class BridgeRegistry {
    public static let shared = BridgeRegistry()
    
    private var handlers: [String: BridgeHandler] = [:]
    
    private init() {}
    
    public func initialize() {
        handlers.removeAll()
    }
    
    public func register(_ bridgeName: String, handler: BridgeHandler) {
        handlers[bridgeName] = handler
        print("BridgeRegistry: Registered bridge: \(bridgeName)")
    }
    
    public func unregister(_ bridgeName: String) {
        handlers.removeValue(forKey: bridgeName)
        print("BridgeRegistry: Unregistered bridge: \(bridgeName)")
    }
    
    public func get(_ bridgeName: String) -> BridgeHandler? {
        return handlers[bridgeName]
    }
}
